### ISSUE TITLE
Orphan InfoWindow when combined with Clusterer

### ISSUE THREAD

**Issue by Insider32** [JustFly1984/react-google-maps-api#193]
I have set up my map component so as to only have 1 InfoWindow at any time, by tracking the clicked/selected marker by a unique ID.

However, I am facing the following issue when the above is combined with the Clusterer.
- I click on a Marker and the InfoWindow appears normally.
- I zoom out, and that Marker is now part of a cluster. The InfoWindow is not visible at this state, but if you zoom in, it re-appears once again, which is desirable.
- On the zoomed-out state, where the above InfoWindow is not visible as the Marker has been clustered, I now click on another single Marker to open its InfoWindow.
- The newly-clicked Marker's InfoWindow opens as normal, however upon zooming in again to check if the other Marker's InfoWindow is still open or not, I noticed that the actual contents have vanished from the InfoWindow and there is some remnant div from google maps. Clicking that marker again shows the actual InfoWindow again but the blank infowindow remains.

You can see the above "phases" in the image below:

![image](https://user-images.githubusercontent.com/3471403/61717083-119a3180-ad69-11e9-9e77-54a4758d06cf.png)

I will try to have a demo with the issue reproduced asap. In the meantime has anyone faced this issue before?
It looks similar to https://github.com/JustFly1984/react-google-maps-api/issues/86 which has apparently been resolved though.

node --version: 10.16.0

react version: 16.6.3

@react-google-maps/api version: 1.4.2

**Comment by Insider32** [comment]
Added repro: https://codesandbox.io/s/sparkling-water-ds2o7

**Comment by georgiosd** [comment]
@JustFly1984 any help you can give us on the issue will be greately appreciated 💯

**Comment by georgiosd** [comment]
I think we have narrowed down the problem to the clusterer removing the markers but not the info window, or all of it. The infowindow wrapper stays dangling around, not even attached to the DOM.

**Comment by JustFly1984** [comment]
I think marker clusterer requires a rewrite to functional Components

**Comment by georgiosd** [comment]
@JustFly1984 that sounds great of course, but would you know of any workarounds for the moment?

**Comment by JustFly1984** [comment]
@georgiosd 
I do not have time to debug this weekend. If you have time, your PR is welcome.
You can make an example in gatsby-example for better debug experience.

**Comment by r007** [comment]
+1 same issue.

**Comment by JustFly1984** [comment]
@r007 Sorry I do not have time to debug. Your PR is welcome. You can fork our project, run it locally, build it and make new example in gatsby-example.

We have cloned marker clusterer as it is, and refactored it to typescript, so this bugs possible came from original version.

**Comment by leandrosalgado** [comment]
I've come across the same bug and it always happens after the user changes the zoom.

 I solved it writing my custom hook useMarkerClusterer :

https://codesandbox.io/s/suspicious-ramanujan-hiwj0

P.S.: The code could be improved in many ways, for example, I'm using the hook useGoogleMap, it could receive the map as a parameter.

P.S.2: I'm using lodash in my project that's why I added it as a dependency and I was a bit lazy to change the code.
